pkg/southbound/gnmi/conn_event.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+// ConnEvent connection event
+type ConnEvent struct {
+	Conn      Conn
+	EventType ConnEventType
+}
+
+// ConnEventType a connection event type
+type ConnEventType int
```

Comment:
This is the event for adding/removing connections. I guess we need both. First we need to find out when a SB connection is added to initiate a go routine to listen to connection state changes otherwise we don't know that.

Reply:
Yeah you’re right they should be kept separate